### PR TITLE
possible fix for failing translations upload

### PR DESCRIPTION
**Pull request by czue**
looks to be a new edge case related to https://github.com/dimagi/commcare-hq/pull/7182 and associated PRs.

I'm not totally sure what the expected edge cases are or what a good test should be. Hoping that @emord @orangejenny or @kaapstorm can take a look since they have worked on this problem in the past.

### CODE REVIEW

File: corehq/apps/app_manager/tests/test_translations.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+class AppManagerTranslationsTest(SimpleTestCase):
+
+    def test_escape_output_value(self):
+        test_cases = [
+            ('hello', '<value>hello</value>'),
+            (u'क्लिक', '<value>&#2325;&#2381;&#2354;&#2367;&#2325;</value>'),
+            ('&#39', '<value>&amp;#39</value>'),
+            ('Here is a ref <output value="/data/no_media" /> with some trailing text and bad < xml.',
+             '<value>Here is a ref <output value="/data/no_media"/> with some trailing text and bad &lt;xml.</value>')
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_translations.py]: E501 line too long (118 > 115 characters)

File: corehq/apps/app_manager/tests/test_translations.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# coding=utf8
+from lxml import etree
+from django.test import SimpleTestCase
+from corehq.apps.app_manager.translations import escape_output_value
+
+
+class AppManagerTranslationsTest(SimpleTestCase):
+
+    def test_escape_output_value(self):
+        test_cases = [
```
Comment (czue) [inline corehq/apps/app_manager/tests/test_translations.py]: would be nice to create a comprehensive set of tests that we think we care about. I wasn't sure I had enough context.

File: corehq/apps/app_manager/tests/test_translations.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+            ('hello', '<value>hello</value>'),
+            (u'क्लिक', '<value>&#2325;&#2381;&#2354;&#2367;&#2325;</value>'),
+            ('&#39', '<value>&amp;#39</value>'),
```
Comment (czue) [inline corehq/apps/app_manager/tests/test_translations.py]: this is the one causing the bug

File: corehq/apps/app_manager/tests/test_translations.py (hunk 1 above)
Comment (czue) [inline corehq/apps/app_manager/tests/test_translations.py]: the whitespace after the `&lt;` gets swallowed. wasn't sure if that was intentional or not

File: corehq/apps/app_manager/translations.py [hunk 4]
```diff
@@ -569,11 +574,16 @@ def update_form_translations(sheet, rows, missing_cols, app):
...
+def escape_output_value(value):
+    try:
+        return etree.fromstring(u"<value>{}</value>".format(
+            re.sub("(?<!/)>", "&gt;", re.sub("<\s*(?!output)", "&lt;", value))
+        ))
+    except XMLSyntaxError:
+        # if something went horribly wrong just don't bother with escaping
+        element = Element('value')
+        element.text = value
+        return element
```
Comment (czue) [inline corehq/apps/app_manager/translations.py]: wasn't sure if this was reasonable fallback behavior

**Comment by czue** [comment]
:white_check_mark: is a good sign...

**Comment by czue** [comment]
no blockers on my side so taking off label, but only someone with context should merge :)

File: corehq/apps/app_manager/tests/test_translations.py (hunk 1 above)
Comment (orangejenny) [inline corehq/apps/app_manager/tests/test_translations.py]: not intentional

File: corehq/apps/app_manager/tests/test_translations.py (hunk 2 above)
Comment (orangejenny) [inline corehq/apps/app_manager/tests/test_translations.py]: agreed! thanks for getting it started.

File: corehq/apps/app_manager/translations.py (hunk 4 above)
Comment (orangejenny) [inline corehq/apps/app_manager/translations.py]: Seems reasonable to me - it'll end up escaping everything, so the form will have undesired HTML entities in it, but the only alternative I can think of is to blank out the value, which seems worse (since you'd lose any indication of what the value used to be).

Ideally it'd fail and report something useful back (and be able to handle multiple errors in the same file) but that's probably a longer-term issue.

**Comment by emord** [comment]
Looks good to merge from my side. whitespace issue isn't intentional, but doesn't seem awful or a new regression. Will let @orangejenny merge when done reviewing
